python/examples/tools/custom/openlibrary.py

```diff
@@ -0,0 +1,63 @@
+import asyncio
+from typing import Any
+
+import requests
+from pydantic import BaseModel, Field
+
+from beeai_framework.tools import ToolInputValidationError
+from beeai_framework.tools.tool import Tool
+
+
+class OpenLibraryToolInput(BaseModel):
+    title: str | None = Field(description="Title of book to retrieve.", default=None)
+    olid: str | None = Field(description="Open Library number of book to retrieve.", default=None)
+    subjects: str | None = Field(description="Subject of a book to retrieve.", default=None)
+
+
+class OpenLibraryToolResult(BaseModel):
+    preview_url: str
+    info_url: str
+    bib_key: str
+
+
+class OpenLibraryTool(Tool[OpenLibraryToolInput]):
+    name = "OpenLibrary"
+    description = """Provides access to a library of books with information about book titles,
+        authors, contributors, publication dates, publisher and isbn."""
+    input_schema = OpenLibraryToolInput
+
+    def _run(self, input: OpenLibraryToolInput, _: Any | None = None) -> None:
```

Comment:
```suggestion
    def _run(self, input: OpenLibraryToolInput, _: Any | None = None) -> OpenLibraryToolResult:
```

Reply:
Good catch missed updating this as I changed to using `OpenLibraryToolResult.`